File: selenium/python/DriverFactory.py
import os

from selenium import webdriver

from enum import Enum, auto


class Browser(Enum):
    CHROME = auto()
    EDGE = auto()
    FIREFOX = auto()


class DriverFactory:

    @classmethod
    def _create_chrome_driver(cls):
        return webdriver.Chrome()

    @classmethod
    def _create_firefox_driver(cls):
        return webdriver.Firefox()

    @classmethod
    def _create_edge_driver(cls):
        return webdriver.Edge()

    @classmethod
    def create_driver(cls, browser):
        os.environ['WDM_LOG_LEVEL'] = '0'
        match browser:
            case Browser.CHROME:
                return cls._create_chrome_driver()
            case Browser.EDGE:
                return cls._create_edge_driver()
            case _:
                return cls._create_firefox_driver()

File: selenium/java/lib/DriverFactory.java
package lib;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class DriverFactory {

    public static WebDriver createDriver(Browser browser) {
        switch (browser) {
            case FIREFOX:
                return createFirefoxDriver();
            case EDGE:
                return createEdgeDriver();
            case CHROME:
            default:
                return createChromeDriver();
        }
    }

    private static WebDriver createChromeDriver() {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--no-sandbox");
        options.addArguments("--disable-search-engine-choice-screen");
        return new ChromeDriver(options);
    }

    private static WebDriver createEdgeDriver() {
        return new EdgeDriver();
    }

    private static WebDriver createFirefoxDriver() {
        return new FirefoxDriver();
    }

}
